1.2/percentile_test.go

```diff
@@ -0,0 +1,64 @@
+package percentile
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestGeneratePercentile(t *testing.T) {
+
+	var tests = []struct {
+		input float32
+		p     uint
+	}{
+		{.05, 0},
+		{.3, 15},
+		{.4, 20},
+		{.5, 20},
+		{1.0, 50},
+	}
+	data := []uint{15, 40, 35, 20, 50, 0}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("PASS"), func(t *testing.T) {
+			if output, err := generatePercentile(test.input); err != nil {
+				t.Errorf("Got error, exprected no error")
+			} else if output(data) != test.p {
+				t.Errorf("Fail expected: %v", test.p)
```

Comment:
In line with the other remarks about diagnostic output, we probably want to show both got and expected.